### DISCUSSION TITLE
How to use a space separator without quotes

### DISCUSSION THREAD

**Discussion by DefinitelyADev** [bilal-fazlani/commanddotnet#473] [Q&A] [answered]
I have the folowing class that implements `IArgumentModel`
```cs
public class MyArgs : IArgumentModel
{
    [Option("exclude", Description = "A space-separated list of names ...", Split = ' ')]
    public string[]? Exclude { get; set; }
}
```

When I invoked it I had to invoke it with quotes e.g. `.\mytool.exe mycommand --exclude "this that"`
How can I change it so that it accepts something like e.g. `.\mytool.exe mycommand --exclude this that`

**Comment by drewburlingame** [comment, marked as answer]
You've run into a limitation between terminals/consoles/shells and the programs they run.  The shell will have already split the line on spaces so it can map the command to the program you've developed.  The `args` in `Main(string[] args)` is what was already split.  Given space is the default argument delimiter, the shell (and even the CommandDotNet framework) does not have enough context to understand when a user would **not** want that behavior.   Since this is already split before the program is invoked, you'll have this problem with any framework.

If you really want to try to work around this, you can get access to the original string from Environment.CommandLine but you'll need to parse the entire line yourself.  It's not as trivial as you'd initially think.

Alternatively, do not rely on spaces.  Using `[Options(Split=',')` as [described here](https://commanddotnet.bilal-fazlani.com/arguments/argument-collections/#option-collections), you could enter `.\mytool.exe mycommand --exclude this,that` and that's as close as we can get to what you're looking for.
